EiDiA_frontend/src/components/exportView/ExportMainView.js

```diff
@@ -0,0 +1,417 @@
+        this.setSelectedVariable = this.setSelectedVariable.bind(this);
+        this.selectTemplate = this.selectTemplate.bind(this);
+        this.toggleDialog = this.toggleDialog.bind(this);
+        this.extractVariables = this.extractVariables.bind(this);
+        this.saveTemplate = this.saveTemplate.bind(this);
+        this.addSelectedDocumentToList = this.addSelectedDocumentToList.bind(this);
+        this.mapDocumentsWithVariables = this.mapDocumentsWithVariables.bind(this);
+        this.getTextFromEditorState = this.getTextFromEditorState.bind(this);
+        this.downloadDocument = this.downloadDocument.bind(this);
+        this.setValueToVariable = this.setValueToVariable.bind(this);
+
+        // reference to document editor, allows access to focus() method (focus editor)
+        // necessary for setting inline styles
+        this.docEditor = React.createRef();
+
+        // Methods specific for each subcomponent in RightSidePanel
+        this.actionSet = {
+            [pageNames.selectTemplate]: {
+                onAction1_1: this.selectTemplate,
+                onAction1_2: this.toggleBlockType,
+                onAction2_1: this.addSelectedDocumentToList,
+                onAction3_1: this.props.changeView,
+                onAction3_2: this.setSelectedVariable,
+                onAction3_3: null
+            },
+            [pageNames.editTemplate]: {
+                onAction1_1: this.toggleInlineStyle,
+                onAction1_2: this.toggleBlockType,
+                onAction2_1: null,
+                onAction3_1: this.toggleDialog,
+                onAction3_2: this.setSelectedVariable,
+                onAction3_3: null
+            },
+            [pageNames.edit]: {
+                onAction1_1: this.toggleInlineStyle,
+                onAction1_2: this.toggleBlockType,
+                onAction2_1: this.addSelectedDocumentToList,
+                onAction3_1: this.toggleDialog,
+                onAction3_2: this.setSelectedVariable,
+                onAction3_3: this.setValueToVariable
+            }
+        }
+    }
+
+    componentDidMount() {
+        HttpService.get(endpoints.getTemplateList, (resp) => {
+            let initTemplate = resp.response[0]
+            this.selectTemplate(initTemplate.name, initTemplate.id);
+        }, (err) => {
+            console.log(err);
+        })
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        if (this.props.currentPage !== prevProps.currentPage && this.props.currentPage === pageNames.edit) {
+            this.mapDocumentsWithVariables();
+        }
+    }
+
+    // Set Value to variable manually (not from document)
+    setValueToVariable(value) {
+        let newState = this.state;
+        let variableState = newState.variables;
+        let index = variableState[newState.selectedVariable].index;
+        variableState[newState.selectedVariable].value = value;
+        variableState[newState.selectedVariable].source = value;
```

Comment:
```suggestion
        variableState[newState.selectedVariable].source = 'user';
```
make it obvious that the user entered this value